packages/dds/merge-tree/src/test/client.applyStashedOpFarm.spec.ts

```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    // regenerate the ops that were applied as stashed ops. this simulates resubmit()
+    const regeneratedStashedOps = [];
+    let stashedOpSeq = startingSeq;
+    while (stashedOps.length > 0) {
+        const op = stashedOps.shift();
+        const newMsg = stashClients[op[2]].makeOpMessage(
+            stashClients[op[2]].regeneratePendingOp(
+                op[0],
+                op[1],
+            ));
+
+        regeneratedStashedOps.push(newMsg);
+    }
+
+    // apply the regenerated stashed ops
+    for (const msg of regeneratedStashedOps) {
+        msg.sequenceNumber = ++stashedOpSeq;
+        stashClients.forEach((c) => c.applyMsg(msg));
+    }
+    // all stash and normal clients should now be in the same state,
+    // except #1 (normal) which still has local changes
+    new TestClientLogger([...clients.filter((_, i) => i !== 1), ...stashClients]).validate();
+
+    // regenerate ops for client #1
+    const reconnectMsgs: [ISequencedDocumentMessage, SegmentGroup | SegmentGroup[]][] = [];
+    reconnectClientMsgs.forEach((opData) => {
+        const newMsg = clients[1].makeOpMessage(
+            clients[1].regeneratePendingOp(
+                opData[0],
+                opData[1],
+            ));
+        newMsg.minimumSequenceNumber = minSeq;
+        // apply message doesn't use the segment group, so just pass undefined
+        reconnectMsgs.push([newMsg, undefined]);
+    });
+
+    // apply regenerated ops as stashed ops for client #1
+    for (const messageData of reconnectMsgs) {
+        const localMetadata = stashClients[1].applyStashedOp(messageData[0].contents);
+        stashedOps.push([messageData[0].contents, localMetadata, 1]);
+    }
+    // now both clients at index 1 should be the same
+    new TestClientLogger([clients[1], stashClients[1]]).validate();
+
+    // apply the regenerated ops from client #1
+    while (reconnectMsgs.length > 0) {
+        const [message] = reconnectMsgs.shift();
+        message.sequenceNumber = ++seq;
+        clients.forEach((c) => c.applyMsg(message));
+    }
+
+    // resubmit regenerated stashed ops
+    const reRegeneratedStashedMessages = [];
+    for (const stashedOp of stashedOps) {
+        const newMsg = stashClients[1].makeOpMessage(
+            stashClients[1].regeneratePendingOp(
+                stashedOp[0],
+                stashedOp[1],
+            ));
+        reRegeneratedStashedMessages.push(newMsg);
+    }
+
+    for (const reRegeneratedStashedOp of reRegeneratedStashedMessages) {
+        reRegeneratedStashedOp.sequenceNumber = ++stashedOpSeq;
+        stashClients.forEach((c) => c.applyMsg(reRegeneratedStashedOp));
+    }
+
+    // all clients should now be the same
+    new TestClientLogger([...clients, ...stashClients]).validate();
+
+    return seq;
+}
+
+export const defaultOptions: IMergeTreeOperationRunnerConfig & { minLength: number, clients: IConfigRange } = {
+    minLength: 16,
+    clients: { min: 2, max: 8 },
```

Comment:
min should be at least 3 so you have unmodifed, reconnection, and stashed at a minimum